src/core/lib/transport/promise_endpoint.cc

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2022 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <grpc/support/port_platform.h>
+
+#include "src/core/lib/transport/promise_endpoint.h"
+
+#include <utility>
+
+#include <grpc/support/log.h>
+
+namespace grpc {
+
+namespace internal {
+
+PromiseEndpoint::PromiseEndpoint(
+    std::unique_ptr<grpc_event_engine::experimental::EventEngine::Endpoint>
+        endpoint,
+    grpc_core::SliceBuffer already_received)
```

Comment:
Whats the purpose of this already_received SliceBuffer parameter ?

Reply:
Please refer to the design doc ([internal link](https://docs.google.com/document/d/1vRy0yse-d1heLQRmLPo_0figsTPXJAnNN84tBCAne_s/edit?resourcekey=0-JvUPdq0LaZq8gMkgT9Pzlw#heading=h.qgvc5vr55ytg)).